src/platforms/wayland/displayclient.cpp

```diff
@@ -294,17 +308,22 @@ void mgw::DisplayClient::Output::done(void* data, struct wl_output* /*wl_output*
 
 void mgw::DisplayClient::Output::for_each_display_buffer(std::function<void(DisplayBuffer & )> const& f)
 {
-    if (!window)
+    if (!shell_surface)
     {
-        static wl_shell_surface_listener const shell_surface_listener{
-            [](void*, wl_shell_surface* shell_surface, uint32_t serial){ wl_shell_surface_pong(shell_surface, serial); },
-            [](void*, wl_shell_surface*, uint32_t, int32_t, int32_t){},
-            [](void*, wl_shell_surface*){}
+        static xdg_surface_listener const shell_surface_listener{
+            [](void*, auto xdg_surface, auto serial) { xdg_surface_ack_configure(xdg_surface, serial); },
         };
+        shell_surface = xdg_wm_base_get_xdg_surface(owner->shell, surface);
+        xdg_surface_add_listener(shell_surface, &shell_surface_listener, this);
 
-        window = wl_shell_get_shell_surface(owner->shell, surface);
-        wl_shell_surface_add_listener(window, &shell_surface_listener, this);
-        wl_shell_surface_set_fullscreen(window, WL_SHELL_SURFACE_FULLSCREEN_METHOD_SCALE, 0, output);
+        static xdg_toplevel_listener const shell_toplevel_listener{
```

Comment:
Should this brace be on its own line? Can't check the style guide because the docs seem to be down right now.

Besides this, LGTM.

Reply:
I'm not sure if/what the style guide says about it, but when we're initializing a value we generally put the brace on the same line as the value (unlike declaring functions, classes and structs)